src/app/static/src/app/components/plots/MapLegend.vue

```diff
@@ -103,8 +105,22 @@
                             valAsProportion = 1 - valAsProportion;
                         }
 
-                        if (val >= 1000) {
-                            val = numeral(val).format("0a")
+                        val = formatOutput(val, format, scale, accuracy)
+
+                        if (typeof(val) === 'number') {
```

Comment:
Sorry I'm really confused here - why aren't we just using the result of `formatOutput`?

Reply:
I'm following what was there before. I believe format `0a` was being used to make the numbers a little smaller on the scale (eg, make 10000 into 10k). It wouldn't be appropriate to do that on the table or tooltips but for the scale it makes sense. I've set it up so that it does the scaling/formatting of the `formatOutput` util and then applies the `0a`/`0.0a` formatting on top of that for the very large numbers.

Reply:
Again, sometimes issues with the metadata such an accuracy that is too high will interfere here causing all the numbers to be rounded down to 0. Fixing the metadata should prevent this from happening.

Reply:
I think it might be a good idea not to pass an accuracy parameter into formatOutput here. We're already rounding because of format("0a") so no need to round twice. This would avoid the scale being affected by some of the weird numbers coming in from the metadata as well.

Reply:
Ok, got you.

I think that you could simplify a little here with:

```
if (typeof(val) === "string" && !val.includes('%')) {
     val = parseFloat(val)
}
if (typeof val == "number") {
    ... formatting logic
}

```